### ISSUE TITLE
LLD with --orphan-handling=warn and custom linker script complains about ELF debug sections

### ISSUE THREAD

**Issue by royger** [llvm/llvm-project#54194]
Hello,

It seems like LLD when used with a linker scripts complains about .symtab, .shstrtab and .strtab not explicitly handled in the script:
```
ld: warning: <internal>:(.symtab) is being placed in '.symtab'
ld: warning: <internal>:(.shstrtab) is being placed in '.shstrtab'
ld: warning: <internal>:(.strtab) is being placed in '.strtab'
```
GNU ld doesn't complain about such sections not being placed, and it's unknown which effect adding those sections to the linker script will have. Is it intended for LLD to complain about such sections not being explicitly handled in linker scripts?

**Comment by MaskRay** [comment]
In GNU ld, most linker created sections have the orphan section diagnostic. .symtab .strtab, .shstrtab appear to be special cased and they do not have the orphan section diagnostic. ld.lld's behavior is more consistent and reports the diagnostic even for .symtab/.strtab/.shstrtab. I think the script should be explicit about these three sections as well, not relying on the quirky GNU ld behavior. https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a840c4de56
